src/OneWire.h

```diff
@@ -0,0 +1,203 @@
+
+// Select the table-lookup method of computing the 8-bit CRC
+// by setting this to 1.  The lookup table enlarges code size by
+// about 250 bytes.  It does NOT consume RAM (but did in very
+// old versions of OneWire).  If you disable this, a slower
+// but very compact algorithm is used.
+#define ONEWIRE_USE_CRC8_TABLE 1
+
+// micro controller feature to use internal pull-ups
+// NOTE: may work, depends on controler and bus-load -> analyse signal traces with scope for reassurance
+#define ONEWIRE_USE_PULL_UP  0
+
+// to make the bus more secure you can disable parasitic bus powering
+// and switch to open drain only, just keep in mind to use the
+// send-routines without power=true
+// NOTE: you need to power the devices yourself, because the external
+// PU-resistor might not do it completely
+
+#include "platform.h"
+
+class OneWire {
+private:
+
+    io_reg_t pin_bitMask;
+    volatile io_reg_t* pin_baseReg;
+
+    /// search state
+    uint8_t search_rom_array[8];
+    uint8_t search_last_discrepancy;
+    bool    search_last_device_flag;
+
+public:
+
+    /// ONEWIRE COMMANDS
+    static const uint8_t   CMD_SEARCH_ROM              { 0xF0 };
+    static const uint8_t   CMD_CONDITIONAL_SEARCH      { 0xEC };
+    static const uint8_t   CMD_SKIP_ROM                { 0xCC };
+    static const uint8_t   CMD_SKIP_ROM_OVERDRIVE      { 0x3C };
+    static const uint8_t   CMD_MATCH_ROM               { 0x55 };
+    static const uint8_t   CMD_MATCH_ROM_OVERDRIVE     { 0x69 };
+    static const uint8_t   CMD_READ_ROM                { 0x0F };
+    static const uint8_t   CMD_READ_ROM_OLD            { 0x33 };
+    static const uint8_t   CMD_ALARM_SEARCH            { 0xEC };
+    static const uint8_t   CMD_RESUME                  { 0xA5 };
+    // TODO: should be constexpr, but NRF5x (redbear, ardu primo) have old toolchain
+
+    /// constructor for the bus master
+    ///
+    /// \param pin  number of the pin as stated by controller-framework
+    explicit OneWire(uint8_t pin);
+
+    /// instantiate a default destructor
+    ~OneWire() = default;                             // nothing special to do here
+
+    OneWire(const OneWire& ow) = delete;             // disallow copy constructor
+    OneWire& operator=(OneWire& ow) = delete;        // disallow copy assignment
+    OneWire& operator=(const OneWire& ow) = delete;  // disallow copy assignment
+
+#if (ONEWIRE_GCC_VERSION>40900) // is needed because of an old tool chain for nrf5x
+    OneWire(OneWire&& ow) = default;               // default move constructor
+    OneWire& operator=(OneWire&& ow) = delete;       // disallow move assignment
+#endif
+
+    /// Perform a 1-Wire reset cycle. Will wait up to 250uS for the bus to come high,
+    /// if it doesn't then it is broken or shorted
+    ///
+    /// \return     true = a device responds with a presence pulse.
+    ///             false = there is no device or the bus is shorted or otherwise held low for more than 250uS
+    bool reset();
```

Comment:
This has the same problem as the original version (#79). You can't tell the difference between no devices and a shorted bus.

Reply:
ok, it could be changed easily but brings one more comparison. and it will break old code. most codebases will only test for non-zero. On the other hand - a major version change should be allowed to break code (with proper documentation).
the most elegant solution (so old code still works) would be to add a register for error-codes that could be checked.